### PR TITLE
[docs] Updated Card a11y guide and example

### PR DESCRIPTION
**Pull request by sheila-semrush**
## Motivation and Context

Replaced the "Close" icon with a button in one of the Card examples.
Added a recommendation about using buttons instead of interactive icons to the A11y tab.
Also sending a related MR in Figma to review.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Nice improve.

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly or it's not required.
- [ ] Unit tests are not broken.
- [ ] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

File: website/docs/components/card/examples/complex_example.tsx
Comment (j-mnizhek) [inline website/docs/components/card/examples/complex_example.tsx:1]: Noticed that our Hint sticks to button when we add it to its addon. Highlighted this in our inner channel to discuss.
image

Comment (sheila-semrush) [reply website/docs/components/card/examples/complex_example.tsx:1]: Tried resolving it, but the only way to make it look proper is invalid code which doesn't build. I'm reverting to the incorrect positioning for now, and creating a bug to fix this.

**Review by j-mnizhek — APPROVED** [review]
Finally we have buttons instead of icons there 🥰

File: website/docs/components/card/examples/complex_example.tsx
Comment (sheila-semrush) [reply website/docs/components/card/examples/complex_example.tsx:1]: I've made it look properly, but the code is formally invalid. I've created a separate task to fix this.
image

**Pull request merged**